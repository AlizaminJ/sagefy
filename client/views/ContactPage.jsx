import React from 'react'
import { string } from 'prop-types'
import Layout from './components/Layout'
import Icon from './components/Icon'
import ExternalLink from './components/ExternalLink'

export default function ContactPage({ hash }) {
  return (
    <Layout
      hash={hash}
      page="ContactPage"
      title="Contact"
      description="Need help? Contact Sagefy support."
    >
      <section>
        <h1>
          Need help? <Icon i="contact" s="h1" />
        </h1>
        <ul>
          <li>
            <strong>I have a problem with content.</strong>
            <br />
            <a href="/search">
              <Icon i="talk" /> Discuss it in the site
            </a>
            .
          </li>
          <li>
            <strong>I have an idea for content.</strong>
            <br />
            <a href="/create-subject">
              <Icon i="subject" /> Create a new subject
            </a>
            .
          </li>
          <li>
            <strong>I have an idea for the software.</strong>
            <br />
            <strong>I found a bug.</strong>
            <br />
            <ExternalLink href="https://github.com/heiskr/sagefy/issues">
              <Icon i="github" /> Add to Github issues
            </ExternalLink>
            .
          </li>
          <li>
            <strong>I found a security issue.</strong>
            <br />
            <strong>I need help with my account.</strong>
            <br />
            <strong>My copyright has been violated.</strong>
            <br />
            <strong>I&apos;m a media person.</strong>
            <br />
            <a href="mailto:support@sagefy.org">
              <Icon i="contact" /> Send us an email
            </a>
            .
          </li>
        </ul>
        <p>
          <a href="/">
            Go back <Icon i="home" /> home
          </a>
        </p>
      </section>
    </Layout>
  )
}

ContactPage.propTypes = { hash: string.isRequired }
