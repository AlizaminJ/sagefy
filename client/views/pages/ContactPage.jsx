import React from 'react'
import { Link } from 'react-router-dom'
import Icon from '../components/Icon'

export default function ContactPage() {
  return (
    <div className="ContactPage">
      <section>
        <h1>
          Need help? <Icon i="contact" s="xxl" />
        </h1>
        <ul>
          <li>
            <strong>I have a problem with content.</strong>
            <br />
            <Icon i="talk" /> Discuss it in the site.
          </li>
          <li>
            <strong>I have an idea for content.</strong>
            <br />
            <Link to="/suggest">
              <Icon i="suggest" /> Suggest
            </Link>
            .
          </li>
          <li>
            <strong>I have an idea for the software.</strong>
            <br />
            <strong>I found a bug.</strong>
            <br />
            <a
              href="https://github.com/heiskr/sagefy/issues"
              target="_blank"
              rel="noopener noreferrer"
            >
              <Icon i="github" /> Add to Github issues
            </a>
            .
          </li>
          <li>
            <strong>I found a security issue.</strong>
            <br />
            <strong>I need help with my account.</strong>
            <br />
            <strong>My copyright has been violated.</strong>
            <br />
            <strong>I&apos;m a media person.</strong>
            <br />
            <a href="mailto:support@sagefy.org">
              <Icon i="contact" /> Send us an email
            </a>
          </li>
        </ul>

        <p>
          <Link to="/">
            Go back <Icon i="home" /> home
          </Link>
        </p>
      </section>
    </div>
  )
}
