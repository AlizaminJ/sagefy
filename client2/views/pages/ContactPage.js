import React from 'react'

const uvUrl = 'https://sgef.cc/feedback'

export default function ContactPage() {
  return (
    <div id="contact" className="page">
      <h1>Contact Sagefy</h1>
      <ul>
        <li>
          <strong>I have a problem with content.</strong>
          <br />Discuss it in the site, or propose removing it.
        </li>
        <li>
          <strong>I have an idea for content.</strong>
          <br />Discuss it in the site.
        </li>
        <li>
          <strong>I have an idea for the software.</strong>
          <br />Add it to our <a href={uvUrl}>feedback forum.</a>
        </li>
        <li>
          <strong>I found a bug.</strong>
          <br />Add to{' '}
          <a href="https://github.com/heiskr/sagefy/issues">Github issues</a>.
          For security issues,{' '}
          <a href="mailto:support@sagefy.org">send us an email</a>.
        </li>
        <li>
          <strong>My copyright has been violated.</strong>
          <br />
          <a href="mailto:support@sagefy.org">Send us an email</a>.
        </li>
        <li>
          <strong>I need help with my account.</strong>
          <br />
          <a href="mailto:support@sagefy.org">Send us an email</a>.
        </li>
      </ul>
    </div>
  )
}
