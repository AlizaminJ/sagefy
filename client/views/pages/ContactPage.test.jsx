import ContactPage from './ContactPage'

describe('ContactPage', () => {
  it('should render the contact page', () => {
    expect(ContactPage()).toMatchSnapshot()
  })
})
