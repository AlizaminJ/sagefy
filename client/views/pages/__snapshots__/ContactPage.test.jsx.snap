// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactPage should render the contact page 1`] = `
<Layout hash={hash}
  description="Need help? Contact Sagefy support."
  page="ContactPage"
  title="Contact"
>
  <section>
    <h1>
      Need help?
      <Icon
        i="contact"
        s="xxl"
      />
    </h1>
    <ul>
      <li>
        <strong>
          I have a problem with content.
        </strong>
        <br />
        <Link
          replace={false}
          to="/search"
        >
          <Icon
            i="talk"
            s="m"
          />
           Discuss it in the site
        </Link>
        .
      </li>
      <li>
        <strong>
          I have an idea for content.
        </strong>
        <br />
        <Link
          replace={false}
          to="/create-subject"
        >
          <Icon
            i="subject"
            s="m"
          />
           Create a new subject
        </Link>
        .
      </li>
      <li>
        <strong>
          I have an idea for the software.
        </strong>
        <br />
        <strong>
          I found a bug.
        </strong>
        <br />
        <ExternalLink
          href="https://github.com/heiskr/sagefy/issues"
        >
          <Icon
            i="github"
            s="m"
          />
           Add to Github issues
        </ExternalLink>
        .
      </li>
      <li>
        <strong>
          I found a security issue.
        </strong>
        <br />
        <strong>
          I need help with my account.
        </strong>
        <br />
        <strong>
          My copyright has been violated.
        </strong>
        <br />
        <strong>
          I'm a media person.
        </strong>
        <br />
        <a
          href="mailto:support@sagefy.org"
        >
          <Icon
            i="contact"
            s="m"
          />
           Send us an email
        </a>
        .
      </li>
    </ul>
    <p>
      <Link
        replace={false}
        to="/"
      >
        Go back
        <Icon
          i="home"
          s="m"
        />
         home
      </Link>
    </p>
  </section>
</Layout>
`;
